src/macros.jl

```diff
@@ -89,3 +89,308 @@ macro map(ex)
 
     throw(ArgumentError("unable to match the given expression to a known map type"))
 end
+
+
+
+# @System Macro ================================================================
+# Autors: @mforets, @ueliwechsler
+using LinearAlgebra: I
+using MacroTools: @capture
+using InteractiveUtils: subtypes
+export @system
+
+function corresponding_type(abstract_type, fields::Tuple)
+    TYPES = subtypes(abstract_type)
+    TYPES_FIELDS = fieldnames.(TYPES)
+    is_in(x, y) = all([el ∈ y for el in x])
```

Comment:
Allocating the array is not needed using the two-argument form of all:
```suggestion
    is_in(x, y) = all(el -> el ∈ y, x)
```
But more simply put, i think we can use `is_in(x, y) = x ⊆ y` (see the documentation of `Base.issubset`). So we can remove `is_in(x, y)` altogether and write `x ⊆ fields` and  `fields ⊆ x` below.

Reply:
.. or we use `Base.issetequal` :smile: